Тестовое задание для frontend'a

salmasov opened the pull request:
Добавил frontend/app.js - mock-api с регистрацией-логином
Добавил frondenr/readme.md - задание для фронтенда

salmasov committed:
добавил mock-api

salmasov committed:
app.js, старт readme.md

salmasov committed:
app.js, старт readme.md

salmasov committed:
дополнил readme.md описанием эндпоинтов

salmasov committed:
еще дополнил readme.md

salmasov committed:
поправил верстку md

salmasov committed:
добавил полезные ссылки

salmasov committed:
подправил код, сделал стрелочную функцию

archived001 commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+| success | Boolean | статус запроса | true                                   |
+| message | String  | Сообщение      | "Пользователь успешно зарегистрирован" |
+
+### 404 - регистрация не удалась
```
400?

archived001 commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+Если в заголовке не передавать jwt-токен
+
+### 404 - не удалось получить информацию
```
400

archived001 commented on frontend/app.js:
```diff
@@ -0,0 +1,128 @@
+      data: info
+    });
+  } else {
+    res.status(404).json({
```
400

archived001 commented on frontend/app.js:
```diff
@@ -0,0 +1,128 @@
+  const { username, password } = req.body;
+  const isRegistered = users.some(user => user.username == username);
+  if (isRegistered) {
+    res.status(404).json({
```
400

archived001 reviewed (commented):
ну вот историю с 404 поправить и вроде как норм

salmasov committed:
правки по ревью

salmasov commented:
Поправил 404 -> 400 ошибки сверху

hnkisdead commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+| Поле    |   Тип   | Описание       | Значение                               |
+| ------- | :-----: | -------------- | -------------------------------------- |
+| success | Boolean | статус запроса | true                                   |
```
Мне кажется статус кодов хватит для понимания что ок или не ок, зачем ещё status?

salmasov replied:
Выпилил success'ы

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+# Тестовое задание для frontend-разработчика javascript
+
+Задача - реализовать SPA приложение, в котором пользователь может зарегистрироваться и просмотреть личную информацию о себе
```
```suggestion
Задача - реализовать одностраничное приложение (SPA), в котором пользователь может зарегистрироваться и просмотреть личную информацию о себе.
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+Здесь содержится форма авторизации пользователя.
+
+Так же должен быть предусмотрен переход на страницу `/register` в виде кнопки
```
```suggestion
Также должен быть предусмотрен переход на страницу `/register` в виде кнопки.
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+Здесь содержится форма регистрации пользователя.
+
+Так же должен быть предусмотрен переход на страницу `/login` в виде кнопки
```
```suggestion
Так же должен быть предусмотрен переход на страницу `/login` в виде кнопки
```
```suggestion
Также должен быть предусмотрен переход на страницу `/login` в виде кнопки.
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+Здесь содержится информация о пользователе - его аватар, текст о себе, имя пользователя.
+
+Желательно оформить в виде карточки
```
```suggestion
Желательно оформить в виде карточки.
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+3. Устанавливаем зависимости api - в консоли запускаем команду `npm install`
+4. Запускаем api - в консоли запускаем команду `node app.js`
+
+В результате в коноли получаем output
```
```suggestion
В результате в консоли получаем output:
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+Важные замечания:
+
+- По-умолчанию api запускается по адресу `http://localhost:8080`
```
```suggestion
- По умолчанию api запускается по адресу `http://localhost:8080`
```

graberzz commented on frontend/package.json:
```diff
@@ -0,0 +1,22 @@
+    "express-jwt": "^5.3.1",
+    "express-session": "^1.16.2",
+    "jsonwebtoken": "^8.5.1",
+    "mongoose": "^5.6.12",
```
😮

salmasov replied:
Действительно, лишнее

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+Задача - реализовать SPA приложение, в котором пользователь может зарегистрироваться и просмотреть личную информацию о себе
+
+Для простоты разработки данного приложения виде backend-части в данном репозитории представлен файл `app.js`. backend-часть дополнять и изменять `не нужно`. Процесс запуска api на вашей машине описан ниже.
```
```suggestion
Для простоты разработки данного приложения в виде backend-части в данном репозитории представлен файл `app.js`. backend-часть дополнять и изменять `не нужно`. Процесс запуска api на вашей машине описан ниже.
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+Задача - реализовать SPA приложение, в котором пользователь может зарегистрироваться и просмотреть личную информацию о себе
+
+Для простоты разработки данного приложения виде backend-части в данном репозитории представлен файл `app.js`. backend-часть дополнять и изменять `не нужно`. Процесс запуска api на вашей машине описан ниже.
+Авторизация и регистрация на backend-части реализована через `jwt-token`.
```
```suggestion
Авторизация и регистрация на backend-части реализована через [JSON Web Token](https://ru.wikipedia.org/wiki/JSON_Web_Token).
```

graberzz commented on frontend/package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "express-jwt-auth",
```
палево!!! 😱

salmasov replied:
Поменял на frontent-assigment

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+## Технические требования
+
+- Убедитесь, что у вас стоит последняя node LTS версии https://nodejs.org/en/
```
```suggestion
- Убедитесь, что у вас стоит последняя node LTS версии https://nodejs.org/en/
```
```suggestion
- Убедитесь, что у вас стоит Node.js последней LTS версии: https://nodejs.org/en/
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+## Запуск API
+
+1. Убедитесь, что у вас стоит последняя node LTS версии. Ссылка для скачивания: https://nodejs.org/en/
```
дублирование кстати, хотя может чтоб наверняка..... 🤔

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+## Запуск API
+
+1. Убедитесь, что у вас стоит последняя node LTS версии. Ссылка для скачивания: https://nodejs.org/en/
+2. В терминале/консоли открыть папку [frontend/](frontend/)
```
терминал или консоль таки??? в других местах везде консоль употребляется... надо бы чтоб путаницы не возникало, а то мало ли....

salmasov replied:
Оставил консоль

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+1. Во время разработки приложения не стоит изобретать велосипеды. Для компонентов формы и карточек предусмотрена библиотека `Vuetify.js`.
+1. При работе с `Vue.js` следует использовать `Vuex`, `Vue-cli`, `Vue-router`
+1. При перезапуске backend-части приложения список пользователей сбрасывается к стандартному состоянию!
+
```
Добавить пункт про то что написанные тесты будут плюсом

salmasov replied:
Добавил

graberzz commented on frontend/README.md:
````diff
@@ -0,0 +1,179 @@
+Сервер с API стартовал по адресу http://localhost:8080
+```
+
+При выполнении запросов к api в консоли будут возникать логи, удобно для тестирования приложения
````
```suggestion
При выполнении запросов к api в консоли будут возникать логи, удобно для тестирования приложения
```
```suggestion
При выполнении запросов к api в консоли будут возникать логи, удобно для тестирования приложения.
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+## Примечания
+
+1. Реализацию frontend-части следует делать `в отдельной папке`, а не в этом репозитории. Этот репозиторий содержит в себе только backend-часть в виде api.
```
Отдельном репозитории мб?

graberzz commented on frontend/app.js:
```diff
@@ -0,0 +1,128 @@
+  });
+  res.json({
+    success: true,
+    message: "Пользователь успешно зарегестрирован"
```
```suggestion
    message: "Пользователь успешно зарегистрирован"
```

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+
+### 401 - пользователь не авторизован
+
+Если в заголовке не передавать jwt-токен, либо jwt-токен был просрочен - вернется ответ
```
Уточнить про заголовок.

graberzz replied:
Вот тут надо расписать как токен передавать конкретнее

graberzz commented on frontend/app.js:
```diff
@@ -0,0 +1,128 @@
+});
+
+app.post("/register", (req, res) => {
+  const { username, password } = req.body;
```
А валидаций совсем никаких не будет? Сейчас можно например пустые объекты вместо полей передать...

salmasov replied:
Добавил валидацию на длину имени-пароля

graberzz commented:
Возможно стоит уточнить про протухание токена.

Commit:
Update frontend/app.js

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

Commit:
Update frontend/README.md

Co-Authored-By: Ayaz Z. <ayz@smena.space>

salmasov committed:
Правки по ревью

Выпилил success: true из респонсов
Добавил валидацию регистрации, по длине логина - пароля
Фиксанул опечатки

salmasov committed:
Merge branch 'frontend-mock-api' of https://github.com/smenateam/assignments into frontend-mock-api

salmasov committed:
добавил инфу про токен

hnkisdead commented on frontend/README.md:
```diff
@@ -53,12 +53,18 @@
 
+## Запрос с jwt-токеном
+
+Чтобы отправить запрос с токеном нужно в Temporary Headers запроса указать заголовок `Authorization: Bearer ${токен пользователя}`
```
"Чтобы отправить запрос с токеном нужно указать заголовок". Temporary Headers это из Postman, никак к HTTP не относится

graberzz commented on frontend/README.md:
```diff
@@ -0,0 +1,179 @@
+## Запуск API
+
+1. Убедитесь, что у вас стоит последняя node LTS версии. Ссылка для скачивания: https://nodejs.org/en/
+2. В консоли открыть папку [frontend/](frontend/)
```
ссылочка на 404 ведет
```suggestion
2. В консоли открыть папку `frontend`
```

salmasov committed:
пофиксил ссылку, убрал temporary
